### PR TITLE
Dx 2037 restructure html hierarchy

### PR DESCRIPTION
**Pull request by trieutran91**
The HTML restructure aim to create a better user experience by rearranging the content. The content is now wrapping itself so the user does not need to use horizontal scrolling. The mobile view experience has improved due to the utility of the Topbar component from Design Guide combined with the Sidebar component.

### CODE REVIEW

File: _layouts/default.html [hunk 1]
```diff
@@ -69,15 +87,15 @@
                                     </a>
                                 </div>
                                 <!-- {% if site.search.enabled == true %}
-                                <form class="search-container mx-2 my-4 px-3 py-2 d-flex"
```
Comment (asbjornu) [inline _layouts/default.html:72]: We will need to have the search container still on the developer portal, no?

**Review by asbjornu — CHANGES_REQUESTED** [review]
Amazing! There's only two small nits I could find:

1. I think we need to keep the search form as it is in use on the Developer Portal.
2. The state of the hamburger/close button is not correct when clicking a sidebar leaf node (internal anchor link).

With those two fixed I think we can ship. 🚢 :shipit:

File: _layouts/default.html (hunk 1 above)
Comment (trieutran91) [reply _layouts/default.html:72]: I guess we have not discussed the placement of the search bar/container with the implementation of the new sidebar.
The first level of the sidebar is a bit narrow to hold a search bar/container in my opinion. Perhaps place it in the top right corner in desktop view and in the topbar in mobile view?

Comment (asbjornu) [reply _layouts/default.html:72]: Yep, that sounds reasonable.

Comment (asbjornu) [reply _layouts/default.html:72]: Will you do a quick rebase of the branch against `HEAD` of `main` as well?

**Comment by asbjornu** [comment]
Getting close! Just the search form left! :)

**Review by asbjornu — APPROVED** [review]

**Pull request merged**